cmd/prometheus-config-reloader/main.go

```diff
@@ -158,11 +170,15 @@ func main() {
 			w.Write([]byte(`{"status":"up"}`))
 		})
 
-		srv := http.Server{Addr: *listenAddress}
+		srv := &http.Server{}
 
 		g.Add(func() error {
 			level.Info(logger).Log("msg", "Starting web server for metrics", "listen", *listenAddress)
-			return srv.ListenAndServe()
+			return web.ListenAndServe(srv, &web.FlagConfig{
+				WebListenAddresses: &[]string{*listenAddress},
+				WebConfigFile:      webConfig,
+				WebSystemdSocket:   func() *bool { b := false; return &b }(),
```

Comment:
Sorry, I'm not familiar with the expoter toolkit package

What is WebSystemdSocket used for? Why do we always return false?

Reply:
`WebSystemdSocket` use systemd socket activation listeners instead of port listeners, but is only supported on Linux.
So, set the default `*false` here for different OS.

Reply:
Hmm we can leave it to nil too, no?

Reply:
> Hmm we can leave it to nil too, no?
Yes. It can be set to nil.